src/Timers.cpp

```diff
@@ -210,6 +225,218 @@ bool Timers::FindTagInTimerTags(std::string tag, std::string tags)
     return (regex_match(tags, regex));
 }
 
+std::string Timers::ConvertToAutoTimerTag(std::string tag)
+{
+    std::regex regex (" ");
+    std::string replaceWith = "_";
+
+    return regex_replace(tag, regex, replaceWith);
+}
+
+std::vector<AutoTimer> Timers::LoadAutoTimers()
+{
+  std::vector<AutoTimer> autoTimers;
+
+  std::string url; 
+  url = StringUtils::Format("%s%s", vuData.getConnectionURL().c_str(), "autotimer"); 
+
+  std::string strXML;
+  strXML = vuData.GetHttpXML(url);
+
+  TiXmlDocument xmlDoc;
+  if (!xmlDoc.Parse(strXML.c_str()))
+  {
+    XBMC->Log(LOG_DEBUG, "Unable to parse XML: %s at line %d", xmlDoc.ErrorDesc(), xmlDoc.ErrorRow());
+    return autoTimers;
+  }
+
+  TiXmlHandle hDoc(&xmlDoc);
+  TiXmlElement* pElem;
+  TiXmlHandle hRoot(0);
+
+  pElem = hDoc.FirstChildElement("autotimer").Element();
+
+  if (!pElem)
+  {
+    XBMC->Log(LOG_DEBUG, "%s Could not find <autotimer> element!", __FUNCTION__);
+    return autoTimers;
+  }
+
+  hRoot=TiXmlHandle(pElem);
+
+  TiXmlElement* pNode = hRoot.FirstChildElement("timer").Element();
+
+  if (!pNode)
+  {
+    XBMC->Log(LOG_DEBUG, "Could not find <timer> element");
+    return autoTimers;
+  }
+
+  for (; pNode != NULL; pNode = pNode->NextSiblingElement("timer"))
+  {
+
+    std::string strTmp;
+
+    int iTmp;
+    bool bTmp;
+    int iDisabled;
+
+    std::string name;
+    std::string match;
+    std::string enabled;
+    int id;
+    std::string from;
+    std::string to;
+    std::string encoding;
+    std::string searchType;
+    std::string searchCase;
+    std::string avoidDuplicateDescription;
+    std::string searchForDuplicateDescription;
+
+    AutoTimer autoTimer;
```

Comment:
Defining those (well at least the non-trivial) vars outside the for loop generates more performant code as the vars will not be initialized on every iteration.